src/libraries/Microsoft.PowerFx.Core/Syntax/NamedFormula.cs

```diff
@@ -0,0 +1,74 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System.Collections.Generic;
+using Microsoft.PowerFx.Core.Errors;
+using Microsoft.PowerFx.Core.Localization;
+using Microsoft.PowerFx.Core.Parser;
+using Microsoft.PowerFx.Core.Syntax.Nodes;
+using Microsoft.PowerFx.Core.Utils;
+
+namespace Microsoft.PowerFx.Core.Syntax
+{
+    /// <summary>
+    /// This encapsulates a named formula: its original script, the parsed result, and any parse errors.
+    /// </summary>
+    internal class NamedFormula
+    {
+        /// <summary>
+        /// A script containing one or more named formulas.
+        /// </summary>
+        public readonly string Script;
+
+        // The language settings used for parsing this script.
+        // May be null if the script is to be parsed in the current locale.
+        public readonly ILanguageSettings Loc;
+
+        public Dictionary<DName, TexlNode> FormulasResult;
+
+        private List<TexlError> _errors;
```

Comment:
Do we also need a HasErrors/GetErrors function?

Reply:
Added